lib/common/to-string.ts

```diff
@@ -34,3 +34,13 @@ export function patchFuncToString() {
     return originalFunctionToString.apply(this, arguments);
   };
 }
+
+export function patchObjectToString() {
+  const originalObjectToString = Object.prototype.toString;
+  Object.prototype.toString = function() {
+    if (this instanceof Promise) {
```

Comment:
Isn't the reason why Promise shims are in use because Promise is not available in all common browsers today? So wouldn't this line throw
```log
ReferenceError: Can't find variable: Promise
```

for projects & browsers that haven't window.Promise (neither native nor shim)?

Reply:
`zone.js` has provided `Promise` shim, so it will be ok even some browser or other environment doesn't provide native `Promise`